redbeams/src/main/java/com/sequenceiq/redbeams/flow/redbeams/termination/AbstractRedbeamsTerminationAction.java

```diff
@@ -64,29 +64,30 @@ protected void doExecute(RedbeamsContext context, P payload, Map<Object, Object>
 
     @Override
     protected RedbeamsContext createFlowContext(FlowParameters flowParameters,
-        StateContext<RedbeamsTerminationState, RedbeamsTerminationEvent> stateContext, P payload) {
+            StateContext<RedbeamsTerminationState, RedbeamsTerminationEvent> stateContext, P payload) {
         Optional<DBStack> optionalDBStack = dbStackService.findById(payload.getResourceId());
-
         CloudContext cloudContext = null;
         CloudCredential cloudCredential = null;
         DatabaseStack databaseStack = null;
         DBStack dbStack = null;
-        dbStack = optionalDBStack.get();
-        MDCBuilder.buildMdcContext(dbStack);
-        Location location = location(region(dbStack.getRegion()), availabilityZone(dbStack.getAvailabilityZone()));
-        String userName = dbStack.getOwnerCrn().getUserId();
-        String accountId = dbStack.getOwnerCrn().getAccountId();
-        cloudContext = new CloudContext(dbStack.getId(), dbStack.getName(), dbStack.getResourceCrn().toString(), dbStack.getCloudPlatform(),
-                dbStack.getPlatformVariant(), location, userName, accountId);
-
         if (optionalDBStack.isPresent()) {
+            dbStack = optionalDBStack.get();
+            MDCBuilder.buildMdcContext(dbStack);
+            Location location = location(region(dbStack.getRegion()), availabilityZone(dbStack.getAvailabilityZone()));
+            String userName = dbStack.getOwnerCrn().getUserId();
+            String accountId = dbStack.getOwnerCrn().getAccountId();
+            cloudContext = new CloudContext(dbStack.getId(), dbStack.getName(), dbStack.getResourceCrn().toString(), dbStack.getCloudPlatform(),
+                    dbStack.getPlatformVariant(), location, userName, accountId);
             try {
                 Credential credential = credentialService.getCredentialByEnvCrn(dbStack.getEnvironmentId());
                 cloudCredential = credentialConverter.convert(credential);
             } catch (Exception ex) {
                 LOGGER.warn("Could not detect credential for environment: {}", dbStack.getEnvironmentId());
             }
             databaseStack = databaseStackConverter.convert(dbStack);
+        } else {
+            LOGGER.warn("DBStack for {} id is not found in the database, it is possible only if the redbeams process was killed during the execution" +
```

Comment:
I would reformulate, because currently we think it is only possible under one condition but we might prove wrong. 
It is possible only => it seems to be only possible.